src/lib/stores/queue.svelte.ts

```
@@ -0,0 +1,161 @@
+   * Leaves the queue: calls the DELETE API, clears local state, then navigates
+   * back to the event page (or home if eventId is unavailable).
+   */
+  async leave() {
+    const eventId = this.eventId;
+    this.clear();
+    if (!browser) return;
+    await fetch(`/api/events/${eventId}/queue`, { method: 'DELETE' }).catch(() => {});
+    if (eventId) {
+      goto(`/events/${eventId}`);
+    } else {
+      goto('/');
```

ESLint

[error] Unexpected goto() call without resolve().

(svelte/no-navigation-without-resolve)